salt/kubernetes-minion/init.sls

```diff
@@ -62,13 +62,16 @@ kubelet:
     - require:
       - service: kubelet
 
-# note: br_netfilter is not available in some kernels
-#       not sure we really need it...
-#
-#      - kmod: br_netfilter
-#
-#br_netfilter:
-#  kmod.present
+  # TODO: This needs to wait for the node to register, which takes a few seconds.
+  # Salt doesn't seem to have a retry mechanism in the version were using, so I'm
+  # doing a horrible hack right now.
+  cmd.run:
+    - name: |
+        NEXT_WAIT_TIME=0
+        until kubectl --kubeconfig /var/lib/kubelet/kubeconfig uncordon {{ grains['fqdn'] }} || [ $NEXT_WAIT_TIME -eq 10 ]; do
```

Comment:
the only issue with this is that it will actually mark the task as successful, even when it has failed, and timed out.